HOW_TO.md

```diff
@@ -102,23 +103,114 @@ any template .yaml files.
+This section also applies to modifying an existing workload, including
+any template .yaml files.
+1. git clone https://github.com/redhat-performance/benchmark-runner
+2. cd benchmark-runner
+3. Install prerequisites (these commands assume RHEL/CentOS/Fedora):
+   - dnf install python3-pip
+4. Create workload Dockerfile, example:
+   ```
+   FROM quay.io/centos/centos:stream8
+   Shell/Python that run workload
+   Result in Json (redirect to stdout)
+   Wrap Json output with begin/end workload stamp
+   start_stamp='@@~@@START-WORKLOAD@@~@@'
+   end_stamp='@@~@@END-WORKLOAD@@~@@'
+   ```
+5. Upload image to quay.io
+6. Create workload pod yaml, example:
+   ```
+   kind: Pod
+   apiVersion: v1
+   metadata:
+   name: vdbench-pod
+   namespace: default
+   spec:
+   containers:
+   - name: vdbench-pod
+     namespace: default
+     image: quay.io/ebattat/centos-stream8-vdbench5.04.07-pod:latest
+     imagePullPolicy: "Always"
+     volumeMounts:
+      - name: vdbench-pvc
+        mountPath: "/workload"
+        env:
+      - name: BLOCK_SIZES
+   ```
+   output pod example:
+   ```
+   '@@~@@START-WORKLOAD@@~@@'
+   {
+   "workload": "Name",
+   "Run": "1",
+   "Thread": 1,
+   "IOPS": "30"
+   }
+   '@@~@@END-WORKLOAD@@~@@'
+   ```
+7. Benchmark-runner - add workload Template in [benchmark_runner/common/template_operations/templates](benchmark_runner/common/template_operations/templates)
+   1. Create workload directory for example [benchmark_runner/common/template_operations/templates/vdbench](benchmark_runner/common/template_operations/templates/vdbench)
+   2. Create custom_data_template.yaml for example [benchmark_runner/common/template_operations/templates/vdbench/vdbench_data_template.yaml](benchmark_runner/common/template_operations/templates/vdbench/vdbench_data_template.yaml)
+      put here all the data that should be replaced by Jinja in
+   3. Create custom pod template [benchmark_runner/common/template_operations/templates/vdbench/internal_data/vdbench_pod_template.yaml](benchmark_runner/common/template_operations/templates/vdbench/internal_data/vdbench_pod_template.yaml)
+8. Create Workload class [benchmark_runner/workloads/workloads.py](benchmark_runner/workloads/workloads.py)
+   1. Add custom workload method, example:
+   ```
+      @typechecked
+      @logger_time_stamp
+      def vdbench_pod(self, name: str = ''):
+      """
+      This method run vdbench pod workload
+      :return:
+      """
+      if name == '':
+      name = self.vdbench_pod.__name__
+      run = VdbenchPod()
+      run.vdbench_pod(name=name)
+   ```
+   2. Add custom workload class, [benchmark_runner/workloads/vdbench_pod.py](benchmark_runner/workloads/vdbench_pod.py):
+   Please copy the whole class and functionality
+9. Add workload method name (workload_pod/workload_vm) to environment_variables_dict['workloads'] in [benchmark_runner/main/environment_variables.py](benchmark_runner/main/environment_variables.py)
```

Comment:
Again, eventually we want classes to register themselves.

Reply:
Classes should be able to add their workload names to environment_variables.py themselves.